deployment-automation-framework/python_scripts/__main__.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+"""Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ 
+ WSO2 Inc. licenses this file to you under the Apache License,
+ Version 2.0 (the "License"); you may not use this file except
+ in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+
+"""
+
+"""This main script is the first script is being invoked for Deployment
+Automation. Following tasks are performed,
+  * Check which IaaS to be used based on the [environment] env value
+    of deployment.cfg (Current implementation is for OpenStack).
+  * Get environment specific configuration information.
+  * Load cluster node data.
+  * Spawn instances in OpenStack.
+  * Get facter information from instances and update puppet templates.
+TODO 
+  * Implement for EC2, Physical machines, Local
+  * Evaluate and implement docker based deployment automation
+  
+"""
+
```

Comment:
@Suhan - Please run Python code review tool and fix the issues 
some tools are listed in - http://stackoverflow.com/questions/10163307/are-there-tools-available-for-python-source-code-analysis